dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/GitRepositoryDetector.kt

```diff
@@ -40,6 +42,19 @@ internal class GitRepositoryDetector : RepositoryDetector {
 
     // region Internal
 
+    private fun sanitizeUrl(remoteUrl: String): String {
+        return try {
+            val parsedUrl = URL(remoteUrl)
+            if (parsedUrl.userInfo != null) {
```

Comment:
just an open thought: I'm wondering if this covers all the cases of passing credentials in the remote URL. It seems, by looking on https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols that it should. Also [RFC2396](https://www.ietf.org/rfc/rfc2396.txt) specifies user info indeed. The funny thing, that there is a statement in RFC:

```
Some URL schemes use the format "user:password" in the userinfo
   field. This practice is NOT RECOMMENDED, because the passing of
   authentication information in clear text (such as URI) has proven to
   be a security risk in almost every case where it has been used.
```

Alternatively we could use `remoteUrl` field of plugin configuration.

Reply:
I changed the way we do the sanitization. I think we should be safe now but yeah adding an additional field for specifying the `remoteUrl` in the plugin configuration could be another idea.